cpp/ql/src/Diagnostics/FailedExtractions.ql

```diff
@@ -1,22 +1,16 @@
 /**
  * @name Failed extractions
- * @description Gives the command-line of compilations for which extraction did not run to completion.
+ * @description List all files in the source code directory with extraction errors.
  * @kind diagnostic
  * @id cpp/diagnostics/failed-extractions
  */
 
 import cpp
+import FailedExtractions
 
-class AnonymousCompilation extends Compilation {
-  override string toString() { result = "<compilation>" }
-}
-
-string describe(Compilation c) {
-  if c.getArgument(1) = "--mimic"
-  then result = "compiler invocation " + concat(int i | i > 1 | c.getArgument(i), " " order by i)
-  else result = "extractor invocation " + concat(int i | | c.getArgument(i), " " order by i)
-}
-
-from Compilation c
-where not c.normalTermination()
-select c, "Extraction failed for " + describe(c), 2
+from ExtractionError error
+where
+  error instanceof ExtractionUnknownError or
+  exists(error.getFile().getRelativePath())
+select error, "Extracting failed in " + error.getFile() + " with error " + error.getErrorMessage(),
```

Comment:
```suggestion
select error, "Extraction failed in " + error.getFile() + " with error " + error.getErrorMessage(),
```

Looking at the alerts in the internal PR, one query says "Extracting failed ..." while another says "Extraction aborted ...". I prefer the latter.

Reply:
I disagree - the other file talks only about extractions that did not run to completion, and thus `aborted` is the right wording.
This query also talks about recoverable errors that only affect the AST of one line of code.
Talking about `Extraction aborted` in that case would imply that we failed more broadly to extract source code.
At least that's how I read it?

Reply:
I didn't mean that "failed" should be replaced with "aborted". Only that "Extracting failed" should be replaced with "Extraction failed".

Reply:
Sorry, my bad! I totally misunderstood you. That's a great suggestion.